app/src/main/java/ar/com/wolox/android/training/ui/login/LoginFragment.java

```diff
@@ -0,0 +1,117 @@
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.OnClick;
+
+public class LoginFragment extends WolmoFragment<LoginPresenter> implements ILoginView {
+
+    @BindView(R.id.fragment_login_email) TextView mEmail;
+    @BindView(R.id.fragment_login_password) TextView mPassword;
+
+    @Inject
+    public LoginFragment() {}
+
+    @Override
+    public int layout() {
+        return R.layout.fragment_login;
+    }
+
+    @Override
+    public void init() {
+        mEmail.setText(getPresenter().getUserEmail());
+    }
+
+    public void goToHome() {
+        Intent intent = new Intent(getActivity(), HomeActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        getActivity().finish();
+    }
+
+    @OnClick(R.id.fragment_login_login)
+    public void onLogin() {
+        String email = mEmail.getText().toString();
+        String password = mPassword.getText().toString();
+
+        Boolean validateErrors =    this.validateEmptyFields(email, password)
+                                    && this.validateEmailField(email);
+        if (validateErrors) {
+            getPresenter().login(email, password);
+        }
+    }
+
+    private Boolean validateEmptyFields(String email, String password) {
+
+        if (!this.validateEmptyField(email)) {
+            mEmail.setError(ErrorHandler.getErrorMessage(ErrorCode.EMPTY_FIELDS));
+            return false;
+        } else if (!this.validateEmptyField(password)) {
+            mPassword.setError(ErrorHandler.getErrorMessage(ErrorCode.EMPTY_FIELDS));
+            return false;
+        }
+
+        return true;
+    }
+
+    public Boolean validateEmailField(String email) {
+        if (!this.validateEmail(email)) {
+            mEmail.setError(ErrorHandler.getErrorMessage(ErrorCode.INVALID_EMAIL));
+            return false;
+        }
+
+        return true;
+    }
+
+    private Boolean validateEmptyField(String field) {
+        return field.length() > 0;
+    }
+
+    private Boolean validateEmail(String email) {
+        return android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches();
+    }
+
+    @Override
+    public void onLoginSuccess() {
+        goToHome();
+    }
+
+    @Override
+    public void onLoginEmailError(String error) {
+        mEmail.setError(error);
+    }
+
+    @Override
+    public void onLoginPasswordError(String error) {
+        mPassword.setError(error);
+    }
+
+    @OnClick(R.id.fragment_login_signup)
+    public void onClickSignup() {
+        Intent intent = new Intent(getActivity(), SignupActivity.class);
+        startActivity(intent);
+    }
+
+    @OnClick(R.id.fragment_login_terms_conditions)
+    public void onClickTermsAndConditions() {
+        Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setData(Uri.parse("https://www.wolox.com.ar/"));
```

Comment:
Move this to resources or a constant . I prefer the first one , what do you think?